hal/common/critical.c

```diff
@@ -270,7 +270,7 @@ uint32_t core_util_atomic_incr_u32(uint32_t *valuePtr, uint32_t delta)
 }
 
 void *core_util_atomic_incr_ptr(void **valuePtr, ptrdiff_t delta) {
-    return core_util_atomic_incr((uintptr_t)valuePtr, (uintptr_t)delta);
+    return core_util_atomic_incr_u32((uintptr_t)valuePtr, (uintptr_t)delta);
```

Comment:
I'm not sure I understand the question. `core_util_atomic_incr_ptr` operates on bytes, is the code invalid?

It's not returning a pointer to the stack, just the updated value. Since the value being updated is a pointer, the result is also a pointer. Let me know if that is still confusing.